src/vscode_ocaml_platform.ml

```diff
@@ -19,29 +19,42 @@ let activate (extension : ExtensionContext.t) =
      vscode [output] pane for logs *)
   Process.Env.set "OCAML_LSP_SERVER_LOG" "-";
   let open Promise.Syntax in
-  let* sandbox = Sandbox.of_settings_or_detect () in
-  let is_fallback = Option.is_empty sandbox in
-  let sandbox = Option.value sandbox ~default:Sandbox.Global in
-  let+ instance = Extension_instance.make sandbox in
-  (* register things with vscode, making sure to register their disposables *)
+  let instance = Extension_instance.make () in
   ExtensionContext.subscribe extension
     ~disposable:(Extension_instance.disposable instance);
   Extension_commands.register_all_commands extension instance;
   Dune_formatter.register extension instance;
   Dune_task_provider.register extension instance;
-  if
-    is_fallback
-    (* if the sandbox we just set up is a fallback sandbox, we create a pop-up
-       message to offer the user to pick a sandbox they want; note: if the user
-       picks another sandbox in the pop-up, we redo part of work we have just
-       done; this is the case because we can't wait or rely on user to pick a
-       sandbox: they may ignore the pop-up leaving the extension hanging, so we
-       use fallback; w/ a proper detection mechanism, we would redo work in rare
-       cases *)
-  then
-    let (_ : unit Promise.t) = suggest_to_pick_sandbox instance in
+  let sandbox = Sandbox.of_settings_or_detect () in
+  let (_ : unit Promise.t) =
+    let* sandbox = sandbox in
+    let is_fallback = Option.is_empty sandbox in
+    (* register things with vscode, making sure to register their disposables *)
```

Comment:
I think this comment should be moved a couple lines up to where the registration functions are being called.

Reply:
I removed this comment entirely. It only explains the obvious.